frontend/src/liboperator/operator/devices_test.go

```diff
@@ -123,19 +125,57 @@ func TestPoolRegister(t *testing.T) {
 	}
 }
 
+func TestPoolPreRegister(t *testing.T) {
+	p := NewDevicePool()
+	p.CancelPreRegistration("inexistent_device")
+	if err := p.PreRegister(&apiv1.DeviceDescriptor{
+		DeviceId: "d1",
+		GroupId:  "g1",
+		Owner:    "o1",
+		Name:     "n1",
+	}, make(chan bool, 1)); err != nil {
```

Comment:
nit: declaring a device descriptor variable in a single statement makes it easier to follow.

```
dd1 := &apiv1.DeviceDescriptor{
  DeviceId: "d1",
  GroupId:  "g1",
  Owner:    "o1",
  Name:     "n1",
}
if err := p.PreRegister(dd1, make(chan bool, 1)); err != nil {
  t.Fatal("Failed to pre-register: ", err)
}
```

Reply:
done